### PR TITLE
codegen await `generatePages()` + detail labels

### PR DESCRIPTION
**Pull request by mat-app**

### CODE REVIEW

File: packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts [hunk 1]
```diff
@@ -34,11 +34,11 @@ export default class TemplateResolver {
 
                 //find 'useGeneratedQuery' import and replace it with use'queryName's
                 const { listQueryName } = this._introspection ? getQueryNames(this._introspection, this._entity.getName()) : { listQueryName : undefined }
-                const hookName = queryHookName(listQueryName ?? '')
+                const hookName = 'useAllCustomersQuery' //queryHookName(listQueryName ?? '')//TODO:replace this with realn query name
 
                 const transformUseQueryImport = (node: ts.Node, importName: string, queryName: string) => {
                   if(isImportDeclarationWithName(node, importName)) {
-                    return createImportDeclaration(queryName, './generated/graphql');
+                    return createNamedImportDeclaration(queryName, 'src/generated/graphql');
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts:41]: strings like this need to be configurable

File: packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts [hunk 2]
```diff
@@ -34,11 +34,11 @@ export default class TemplateResolver {
 
                 //find 'useGeneratedQuery' import and replace it with use'queryName's
                 const { listQueryName } = this._introspection ? getQueryNames(this._introspection, this._entity.getName()) : { listQueryName : undefined }
-                const hookName = queryHookName(listQueryName ?? '')
+                const hookName = 'useAllCustomersQuery' //queryHookName(listQueryName ?? '')//TODO:replace this with realn query name
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts:37]: @rastiQ, is there any technical problem?

File: packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts (hunk 1 above)
Comment (mat-app) [reply packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts]: configurable generated folder path

File: packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts (hunk 2 above)
Comment (mat-app) [reply packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts]: it was not working and rasto found bug, that template resolver was not provided with introspection..fixed

File: packages/react-lowcode/src/codegen/generation/generators/detail/interface-generator.ts [hunk 3]
```diff
@@ -1,57 +1,54 @@
...
+    );
+  }
+
+  getInterfaceFields(): ts.TypeElement[] {
+    let inputs: ts.TypeElement[] = [];
+    getProperties(this._entity).forEach((property) => {
+      let propertyInput = this.getPropertySignature(property);
+
+      if (propertyInput) {
+        inputs.push(propertyInput);
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/codegen/generation/generators/detail/interface-generator.ts:39]: please use `getProperties().map()` insteado of `[].push()`

File: packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts [hunk 4]
```diff
@@ -742,6 +717,16 @@ export default class MuiDetailGenerator implements DetailGenerator {
     }
   }
 
+  private getLabelValueAttribute(
```
Comment (jozef-slezak) [inline packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts:720]: let's rename to `createLabelAttribute()`

Comment (mat-app) [reply packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts]: renamed

**Pull request merged**